delivery/routes.py

```diff
@@ -34,3 +39,29 @@ def put(self, restaurant_id: int):
         restaurant.id = restaurant_id
         updated_restaurant = self.restaurants.update(restaurant)
         return jsonify(self.schema.dump(updated_restaurant)), 200
+
+
+class LoginEndpoint(MethodView):
+    def __init__(self, users, schema):
+        self.users = users
+        self.schema = schema
+
+    def post(self):
+        login = request.json.get('login', None)
+        password = request.json.get('password', None)
+        user_password = self.users.get_user(login).password
+        try:
+            hashed = bcrypt.kdf(password.encode(), salt=login.encode(), desired_key_bytes=len(login), rounds=len(password))
+        except:
+            raise WrongPassword('Password is required')
+        result = user_password.encode('latin1').decode('unicode_escape').encode('latin1')
```

Comment:
Why such a complex encode/decode chain is needed?
Can it be just .encode('utf-8')?

Reply:
Moreover, seems that encoding is not consistent, several lines above it is just .encode(), here it is a chain of encode and decode

Reply:
fixed